app/controllers/search_controller.rb

```diff
@@ -60,9 +85,81 @@ def reorder_filters(aggs, active_filters)
 
   def validate_q!
     return if params[:advanced]&.strip.present?
+    return if params[:geobox]&.strip.present?
+    return if params[:geodistance]&.strip.present?
     return if params[:q]&.strip.present?
 
     flash[:error] = 'A search term is required.'
     redirect_to root_url
   end
+
+  def validate_geodistance_presence!
+    return unless params[:geodistance]&.strip == 'true'
+
+    geodistance_params = [params[:geodistanceLatitude]&.strip, params[:geodistanceLongitude]&.strip,
+                          params[:geodistanceDistance]&.strip]
+    return if geodistance_params.all?(&:present?)
+
+    flash[:error] = 'All geospatial distance fields are required.'
+    redirect_to root_url
+  end
+
+  def validate_geobox_presence!
+    return unless params[:geobox]&.strip == 'true'
+
+    geobox_params = [params[:geoboxMinLatitude]&.strip, params[:geoboxMinLongitude]&.strip,
+                     params[:geoboxMaxLatitude]&.strip, params[:geoboxMaxLongitude]&.strip]
+    return if geobox_params.all?(&:present?)
+
+    flash[:error] = 'All bounding box fields are required.'
+    redirect_to root_url
+  end
+
+  def validate_geodistance_range!
+    return unless params[:geodistance]&.strip == 'true'
+
+    lat = params[:geodistanceLatitude]&.strip.to_f
+    long = params[:geodistanceLongitude]&.strip.to_f
+    return if lat.between?(-90.0, 90.0)
+    return if long.between?(-180.0, 180.0)
+
+    flash[:error] = 'Latitude must be between -90.0 and 90.0, and longitude must be -180.0 and 180.0.'
+    redirect_to root_url
+  end
+
+  def validate_geobox_range!
+    return unless params[:geobox]&.strip == 'true'
+
+    geobox_lat = [params[:geoboxMinLatitude]&.strip.to_f, params[:geoboxMaxLatitude]&.strip.to_f]
+    geobox_long = [params[:geoboxMinLongitude]&.strip.to_f, params[:geoboxMaxLongitude]&.strip.to_f]
+    return if geobox_lat.all? { |lat| lat.between?(-90.0, 90.0) }
+    return if geobox_long.all? { |long| long.between?(-180.0, 180.0) }
+
+    flash[:error] = 'Latitude must be between -90.0 and 90.0, and longitude must be -180.0 and 180.0.'
+    redirect_to root_url
+  end
+
+  def validate_geodistance_values!
+    return unless params[:geodistance]&.strip == 'true'
+
+    distance = params[:geodistanceDistance]&.strip.to_i
+    return if distance > 0
+
+    flash[:error] = 'Distance must include an integer greater than 0.'
+    redirect_to root_url
+  end
+
+  def validate_geobox_values!
+    return unless params[:geobox]&.strip == 'true'
+
+    geobox_params = [params[:geoboxMinLatitude]&.strip.to_f, params[:geoboxMinLongitude]&.strip.to_f,
+                     params[:geoboxMaxLatitude]&.strip.to_f, params[:geoboxMaxLongitude]&.strip.to_f]
+
+    # Confirm that min values are lower than max values
+    return if geobox_params.any?(&:blank?)
+    return if geobox_params[0] < geobox_params[2] && geobox_params[1] < geobox_params[3]
```

Comment:
That's kind of a relief. This is one of the trickier validations, and I think it would become especially complex when we start adding more of these validations to client-side.